File: launcher/utils/windowsize.py
class WindowSize:

    def __init__(self, width: int = None, height: int = None):
        self.width = None
        self.height = None
        self.fullscreen = False

        self.resize(width, height)

    def resize(self, width: int = None, height: int = None):
        self.width = width
        self.height = height
        self.fullscreen = False

        if not width and not height:
            self.width = 0
            self.height = 0
            self.fullscreen = True

    def fullscreen(self):
        self.resize()

    def to_tuple(self):
        return self.width, self.height

File: src/main/java/morningmc/foliage/minecraft/launch/options/WindowSize.java
package morningmc.foliage.minecraft.launch.options;

public class WindowSize {
    private boolean fullscreen;
    private int width;
    private int height;

    public WindowSize() {
        fullscreen = true;
    }

    public WindowSize(int width, int height) {
        if (width < 0 || height < 0) {
            throw new IllegalArgumentException("width < 0 or height < 0");
        }

        fullscreen = false;
        this.width = width;
        this.height = height;
    }

    public boolean isFullscreen() {
        return fullscreen;
    }

    public int getWidth() {
        return width;
    }

    public int getHeight() {
        return height;
    }
}
